core/src/main/java/org/jruby/RubyException.java

```diff
@@ -302,10 +303,9 @@ public RubyString inspect(ThreadContext context) {
 
         if (exception.isEmpty()) return context.runtime.newString(rubyClass);
 
-        return RubyString.newString(context.runtime,
-                new StringBuilder(2 + rubyClass.length() + 2 + exception.size() + 1).
-                    append("#<").append(rubyClass).append(": ").append(exception.getByteList()).append('>')
-        );
+        String exceptionStartStr = new StringBuilder(2 + rubyClass.length() + 2).append("#<").append(rubyClass).append(": ").toString();
```

Comment:
you should nuke the `StringBuilder` here - it shouldn't be needed

Reply:
I think nuking is done now.

Reply:
the last round-trip (Ruby -> Java -> Ruby string) is redundant, but we can fix that later ...